be/src/vec/functions/function_string.h

```diff
@@ -2292,30 +2292,76 @@ class FunctionStringLocatePos : public IFunction {
 
     Status execute_impl(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
                         size_t result, size_t input_rows_count) override {
-        auto col_substr =
-                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
-        auto col_str =
+        const auto& left = block.get_by_position(arguments[0]);
+        const auto& right =
```

Comment:
maybe the right column seems more likely to be constant? If so, we should optimize for `rcol` but not `lcol` and use `default_preprocess_parameter_columns` to deal with other columns. Then our dispatcher could distinguish by wether `lcol` and `pos` are all const.

Reply:
> maybe the right column seems more likely to be constant? If so, we should optimize for `rcol` but not `lcol` and use `default_preprocess_parameter_columns` to deal with other columns. Then our dispatcher could distinguish by wether `lcol` and `pos` are all const.

locale(subStr, str, pos) , this function parameter is subStr, str, pos.
On one hand,  subStr is likely to be constant.
On the other hand, because of init stringsearch is a waste of time,  so optimize left col is more benefit than the right col.